### PR TITLE
rework stdin handling

### PR DESCRIPTION
**Pull request by whyrusleeping**
no comment

License: MIT
Signed-off-by: Jeromy <why@ipfs.io>

### CODE REVIEW

**Comment by whyrusleeping** [comment]
cc @jbenet @Kubuxu @lgierth

**Comment by whyrusleeping** [comment]
Alright, this should be ready for review. cc @jbenet @Kubuxu @thomas-gardner

File: commands/cli/parse.go [hunk 1]
```diff
@@ -17,6 +17,11 @@ import (
 
 var log = logging.Logger("commands/cli")
 
+// stdinSpecialName is a name applied to the 'stdin' file so we can differentiate
+// it from potential 'real' files being passed in. The '*' character is invalid in
+// path names and won't appear otherwise.
+const stdinSpecialName = "*stdin*"
```
Comment (Kubuxu) [inline commands/cli/parse.go]: `*` is valid in path names.

Comment (Kubuxu) [inline commands/cli/parse.go]:
```
 ../test/sharness   patch-1  echo Yes it works > '*stdin*'
 ../test/sharness   patch-1 ?  cat '*stdin*'
Yes it works
```

Comment (whyrusleeping) [inline commands/cli/parse.go]: youre right, and this is no longer needed.

**Comment by RichardLitt** [comment]
Does this affect the HTTP API at all?

**Comment by jbenet** [comment]
This PR:
- merged commits that do not pass tests (squash them)
- merged commits with useless commit messages... (squash them, or write better commit messages)

**Pull request merged**